tests/gitlab_ci/test_runner.py

```diff
@@ -55,8 +55,25 @@ def test_runner_image_check(self):
         )
         self.assertEqual(len(report.failed_checks), 0)
         self.assertEqual(report.parsing_errors, [])
+        self.assertEqual(len(report.passed_checks), 8)
+        self.assertEqual(report.skipped_checks, [])
+
+    def test_runner_image_resources(self):
+        current_dir = os.path.dirname(os.path.realpath(__file__))
+        valid_dir_path = os.path.join(current_dir, "resources/resource_images")
+        runner = Runner()
+        report = runner.run(
+            root_folder=valid_dir_path,
+            runner_filter=RunnerFilter(framework=['gitlab_ci'], checks=['CKV_GITLABCI_3'])
+        )
+        self.assertEqual(len(report.failed_checks), 0)
+        self.assertEqual(report.parsing_errors, [])
         self.assertEqual(len(report.passed_checks), 4)
         self.assertEqual(report.skipped_checks, [])
+        self.assertEqual(report.passed_checks[0].resource, 'prebuild')
+        self.assertEqual(report.passed_checks[1].resource, 'build.image')
+        self.assertEqual(report.passed_checks[2].resource, 'deploy')
+        self.assertEqual(report.passed_checks[3].resource, 'deploy')
```

Comment:
shouldn't both resource ids be `deploy.services`?

Reply:
Depends on the specific case, but we're providing the most focused resource we can reliably compute. The reason we don't see `deploy.services` is because services in our example is a list, which doesn't have start/end lines, so cannot be found within the encapsulating dict. The image, on the other hand, is a dict which has start/end lines so that's why we output `build.image`. But as I wrote in the description, we'll need further work around this to handle multiple images per resource, which will force us to compute more focused resources, in which case for lists we'll probably compute something like `deploy.services[1]`. 😄